web/instances/instances.go

```diff
@@ -573,6 +576,175 @@ func lsContexts(c echo.Context) error {
 	return c.JSON(http.StatusOK, result)
 }
 
+func appVersion(c echo.Context) error {
+	instances, err := instance.List()
+	if err != nil {
+		return nil
+	}
+	appSlug := c.Param("slug")
+	version := c.Param("version")
+
+	var instancesAppVersion []string
+	var doc app.WebappManifest
+
+	for _, instance := range instances {
+		err := couchdb.GetDoc(instance, consts.Apps, consts.Apps+"/"+appSlug, &doc)
+		if err == nil {
+			if doc.Version() == version {
+				instancesAppVersion = append(instancesAppVersion, instance.Domain)
+			}
+		}
+	}
+
+	i := struct {
+		Instances []string `json:"instances"`
+	}{
+		instancesAppVersion,
+	}
+
+	return c.JSON(http.StatusOK, i)
+}
+
+// contentMismatchFixer fixes the 64k bug
+func contentMismatchFixer(c echo.Context) error {
+	corruptedSuffix := "-corrupted"
+
+	domain := c.Param("domain")
+	inst, err := lifecycle.GetInstance(domain)
+	if err != nil {
+		return fmt.Errorf("Cannot find instance %s", domain)
+	}
+
+	dryRun := true
+	if dryParam := c.QueryParam("dry_run"); dryParam != "" {
+		if dry, err := strconv.ParseBool(dryParam); err == nil {
+			dryRun = dry
+		}
+	}
+
+	// Directly requesting the /fsck endpoint by forging a custom echo.Context
```

Comment:
Yikes, it seems weird to do that. Can't we just call the right method of the VFS?

Reply:
And it would be nice to have the code for the fixers in their own file: `web/instances/fixers.go`